src/containers/forms/OrganizationSelect/index.tsx

```diff
@@ -0,0 +1,171 @@
+} from '../../../store/ducks/opensrp/organizations';
+
+reducerRegistry.register(assignmentReducerName, assignmentReducer);
+reducerRegistry.register(organizationsReducerName, organizationsReducer);
+
+/** react-select Option */
+interface SelectOption {
+  label: string;
+  value: string;
+}
+
+/** OrganizationSelect props */
+export interface OrganizationSelectProps {
+  assignments: Assignment[];
+  fetchAssignmentsAction: typeof fetchAssignments;
+  fetchOrganizationsAction: typeof fetchOrganizations;
+  jurisdictionId: string;
+  name: string; // name of the input
+  organizations: Organization[];
+  planId: string;
+  serviceClass: typeof OpenSRPService /** the OpenSRP service */;
+  values: SelectOption[];
+}
+
+/** default props for OrganizationSelect */
+const defaultProps: OrganizationSelectProps = {
+  assignments: [],
+  fetchAssignmentsAction: fetchAssignments,
+  fetchOrganizationsAction: fetchOrganizations,
+  jurisdictionId: '',
+  name: '',
+  organizations: [],
+  planId: '',
+  serviceClass: OpenSRPService,
+  values: [],
+};
+
+/**
+ * OrganizationSelect - a cascading select for Jurisdictions
+ * Allows you to drill-down Jurisdictions until you select a Focus Area
+ * This is simply a Higher Order Component that wraps around AsyncSelect
+ */
+const OrganizationSelect = (props: OrganizationSelectProps) => {
+  const {
+    fetchAssignmentsAction,
+    fetchOrganizationsAction,
+    jurisdictionId,
+    name,
+    organizations,
+    planId,
+    serviceClass,
+    values: selectOptions,
+  } = props;
+
+  const loadOrganizations = async (service: typeof serviceClass) => {
+    const serve = new service(OPENSRP_ORGANIZATION_ENDPOINT);
+    serve
+      .list()
+      .then((response: Organization[]) => store.dispatch(fetchOrganizationsAction(response)))
+      .catch((err: Error) => {
+        /** TODO - find something to do with error */
+      });
+  };
+
+  useEffect(() => {
+    loadOrganizations(serviceClass);
+  }, []);
+
+  /** Get select options from OpenSRP as a promise */
+
+  /**
+   * onChange callback
+   * unfortunately we have to set the type of option as any (for now)
+   */
+  const handleChange = (nextValues: any, meta: any) => {
+    // handle input change => updatedStore
+    // console.log(nextValues, meta);
+    const nextAssignments: Assignment[] = nextValues.map(
+      (v: SelectOption) =>
+        ({
+          jurisdiction: jurisdictionId,
+          organization: v.value,
+          plan: planId,
+        } as Assignment)
+    );
+    fetchAssignmentsAction(nextAssignments);
+  };
+
+  return (
+    <Select
+      name={name}
+      bsSize="lg"
+      placeholder={'Select'}
+      aria-label={'Select'}
+      onChange={handleChange}
+      defaultOptions={true}
+      options={organizations.map(
+        o =>
+          ({
+            label: o.name,
+            value: o.identifier,
+          } as SelectOption)
+      )}
+      isClearable={true}
+      isMulti={true}
+      values={selectOptions}
```

Comment:
@moshthepitt @p-netm I can't quite get this one to work, when this component mounts and renders it doesn't show the Teams which were already selected. Thoughts/help?

Reply:
@cKellyDesign  could it be that this prop should be spelled `value`,

Reply:
yeah @p-netm good call, this helps quite a bit in a module that also imports `values` from lodash!